backoffice/views.py

```diff
@@ -455,3 +458,183 @@ def perform_box_action(request, vehicle, action_to_perform, user):
+            }
+        if most_recent["doors_locked"] is None and t.doors_locked is not None:
+            most_recent["doors_locked"] = {
+                "value": t.doors_locked,
+                "age": timezone.now() - t.created_at,
+            }
+        if most_recent["miles"] is None and t.odometer_miles is not None:
+            most_recent["miles"] = {
+                "value": t.odometer_miles,
+                "age": timezone.now() - t.created_at,
+            }
+        # Stop looking through telemetry if all most_recent values found
+        if all(value is not None for value in most_recent.values()):
+            break
+
+    for key in most_recent:
+        if most_recent[key] is not None:
+            most_recent[key]["age"] = breakdown_timedelta(most_recent[key]["age"])
+
+    if most_recent["soc"] is not None:
+        most_recent["soc_dial"] = 251.2 * (1 - t.soc_percent / 100)
+    else:
+        most_recent["soc_dial"] = 0
+
+    paginator = Paginator(telemetry, 10)
+
+    page_number = request.GET.get("page", 1)
+    page_obj = paginator.get_page(page_number)
+    page_range = paginator.get_elided_page_range(
+        number=page_number, on_each_side=1, on_ends=1
+    )
+
+    context = {
+        "vehicle": vehicle,
+        "telemetry": telemetry,
+        "most_recent": most_recent,
+        "page": page_obj,
+        "page_range": page_range,
+        "bookings": [last_booking, current_booking, next_booking],
+    }
+    return render(request, "backoffice/vehicles/details.html", context)
+
+
+def breakdown_timedelta(timedelta):
+    seconds_in_an_hour = 60 * 60
+    seconds_in_a_day = 24 * seconds_in_an_hour
+    seconds_in_a_month = 30 * seconds_in_a_day
+    seconds_in_a_year = 365 * seconds_in_a_day
+
+    all_seconds = timedelta.days * seconds_in_a_day + timedelta.seconds
+    years, seconds_remaining = divmod(all_seconds, seconds_in_a_year)
+    months, seconds_remaining = divmod(seconds_remaining, seconds_in_a_month)
+    days, seconds_remaining = divmod(seconds_remaining, seconds_in_a_day)
+    hours, seconds_remaining = divmod(seconds_remaining, seconds_in_an_hour)
+    minutes, seconds_remaining = divmod(seconds_remaining, 60)
+    seconds = int(seconds_remaining)
+    if years > 0:
+        years_str = f"{years} years, "
+    else:
+        years_str = ""
+    if months > 0:
+        months_str = f"{months} months, "
+    else:
+        months_str = ""
+    if days > 0:
+        days_str = f"{days} days, "
+    else:
+        days_str = ""
+    if hours > 0:
+        hours_str = f"{hours} hours, "
+    else:
+        hours_str = ""
+    if minutes > 0:
+        minutes_str = f"{minutes} minutes, "
+    else:
+        minutes_str = ""
+    if seconds > 0:
+        seconds_str = f"and {seconds} seconds"
+    else:
+        seconds_str = ""
+
+    return f"{years_str}{months_str}{days_str}{hours_str}{minutes_str}{seconds_str}"
+
+
+@require_backoffice_access
+def get_telemetry(request):
```

Comment:
This appears to be a POST request. Do we need to validate the CSRF token in this request, or is middleware already doing that somewhere?